scim/data_current_user.go

```diff
@@ -55,6 +59,11 @@ func DataSourceCurrentUser() *schema.Resource {
 			d.Set("user_name", me.UserName)
 			d.Set("home", fmt.Sprintf("/Users/%s", me.UserName))
 			d.Set("repos", fmt.Sprintf("/Repos/%s", me.UserName))
+			if strings.Contains(me.UserName, "@") {
```

Comment:
@gauthamsunjay FYI: this is how we tell if the current principal is a user or service principal. It would be nice to expose this in the SCIM /Me response if it isn't already there.

Reply:
@baishen-db can you take a look here? Is SCIM/Me exposed for users and service principals?

Reply:
It's exposed to both. The main question is how to distinguish between them...

Reply:
I just spoke to @baishen-db and @michaelduan-db offline. `SCIM/Me` doesn't currently distinguish between user and a service principal.